lib/pka.c

```diff
@@ -825,6 +840,16 @@ void* polling_func(void* args)
     do {
        int errors = 0, count = 0;
 
+       uint32_t async_jobs_cnt = pka_atomic32_load(&pka_async_curr_jobs_cnt);
+
+       // Only do async result query here when there is outstanding async request
+       // This check handles the scenarios of mixed async and async in openssl

```

Comment:
Should we check if this is an ASYNC job first?  (NULL != ASYNC_get_current_job())

Reply:
Cannot check it here since the polling thread is no in async Job context.